pallets/issue/src/lib.rs

```diff
@@ -855,4 +823,56 @@ impl<T: Config> Pallet<T> {
 		}
 		Ok(())
 	}
+
+	fn validate_stellar_transaction(
```

Comment:
The refactoring is nice but maybe we should split this function even further. It does not only validate but also extract the amount which is confusing judging from the naming. 

So how about we remove lines 870-876 from here and put it back to the outer function? We would then change this function to then return the `transaction_envelope` so that it can be used in the outer function.

Reply:
updated